### PR TITLE
chrome-gnome-shell: refactor

### PR DESCRIPTION
**Pull request by jtojnar**
The native connector requires a D-Bus service. For firefox it is also necessary to enable the host connector, for example by setting `nixpkgs.config.firefox.enableGnomeExtensions` to `true`.

<del>

Unfortunately, it still does not work because the browsers require manifest files, [Chromium](https://developer.chrome.com/extensions/nativeMessaging) in `/etc/chromium/native-messaging-hosts` and [Firefox](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Native_messaging) in `/usr/lib64/mozilla/native-messaging-hosts`.

One way to make it work is to copy the files to local folder:

```bash
# Firefox
mkdir ~/.mozilla/native-messaging-hosts
cp $(dirname $(readlink -f $(which chrome-gnome-shell)))/../lib/mozilla/native-messaging-hosts/org.gnome.chrome_gnome_shell.json ~/.mozilla/native-messaging-hosts
# Chromium
mkdir ~/.config/chromium/NativeMessagingHosts
cp $(dirname $(readlink -f $(which chrome-gnome-shell)))/../etc/chromium/native-messaging-hosts/org.gnome.chrome_gnome_shell.json ~/.config/chromium/NativeMessagingHosts
```
</del>

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @dyrnade

### CODE REVIEW

File: pkgs/desktops/gnome-3/extensions/chrome-gnome-shell/default.nix [hunk 1]
```diff
@@ -1,31 +1,20 @@
...
-   substituteInPlace cmake_install.cmake --replace "/etc" "$out/etc"
- '';
-
- postInstall = ''
-    rm $out/etc/opt/chrome/policies/managed/chrome-gnome-shell.json
-    rm $out/etc/chromium/policies/managed/chrome-gnome-shell.json
-    wrapProgram $out/usr/bin/chrome-gnome-shell \
-      --prefix PATH : '"${dbus}/bin"' \
-      --prefix PATH : '"${gnome3.gnome_shell}/bin"' \
-      --prefix PYTHONPATH : "$PYTHONPATH"
```
Comment (orivej) [inline pkgs/desktops/gnome-3/extensions/chrome-gnome-shell/default.nix]: Why is the wrapper no longer needed?

Comment (jtojnar) [reply pkgs/desktops/gnome-3/extensions/chrome-gnome-shell/default.nix]: Switched to `wrapGAppsHook`. I still wrap `PYTHONPATH`  through `wrapPrefixVariables`. `PATH` does not seem to be needed.

**Review by orivej — APPROVED** [review]

**Comment by deedrah** [comment]
I think you could get it working in chromium/chrome out of the box by adding this lines into module:
```
  environment.etc."chromium/native-messaging-hosts/org.gnome.chrome_gnome_shell.json".source = "${pkgs.chrome-gnome-shell}/etc/chromium/native-messaging-hosts/org.gnome.chrome_gnome_shell.json";
  environment.etc."opt/chrome/native-messaging-hosts/org.gnome.chrome_gnome_shell.json".source = "${pkgs.chrome-gnome-shell}/etc/opt/chrome/native-messaging-hosts/org.gnome.chrome_gnome_shell.json";
```

**Comment by joachifm** [comment]
@grahamcofborg eval

**Comment by jtojnar** [comment]
@deedrah I merged your firefox patch.

**Pull request merged**